### DISCUSSION TITLE
Minecraft compiled with newer version of Java

### DISCUSSION THREAD

**Discussion by sugarjig** [GameServerManagers/LinuxGSM#4561] [Q&A] [answered]
I'm suddenly getting this error when starting a Minecraft server
```
java.lang.UnsupportedClassVersionError: net/minecraft/bundler/Main has been compiled by a more recent version of the Java Runtime (class file version 65.0), this version of the Java Runtime only recognizes class file versions up to 61.0
```
I'm guessing the latest version of Minecraft was compiled with Java 21, and I have Java 17 installed.

If memory serves correctly, I used LGSM to install dependencies. Will LGSM update its Minecraft install/update scripts to automatically add Java 21 to a system? Or should I update Java by hand?

**Comment by tumekee** [comment]
> I'm suddenly getting this error when starting a Minecraft server
> 
> ```
> java.lang.UnsupportedClassVersionError: net/minecraft/bundler/Main has been compiled by a more recent version of the Java Runtime (class file version 65.0), this version of the Java Runtime only recognizes class file versions up to 61.0
> ```
> 
> I'm guessing the latest version of Minecraft was compiled with Java 21, and I have Java 17 installed.
> 
> If memory serves correctly, I used LGSM to install dependencies. Will LGSM update its Minecraft install/update scripts to automatically add Java 21 to a system? Or should I update Java by hand?

try - sudo apt install openjdk-21-jdk

**Reply by Medienzentrum-Landkreis-Harburg** [reply]
on debian bookworm, just install https://scqr.net/en/blog/2023/10/14/java-21-on-devuan-5-debian-12-install-manually/ openjdk21 and exort the environment variables for both PATH and Home as user who runs the lgsm

**Comment by MicLieg** [comment, marked as answer]
I've already submitted a PR https://github.com/GameServerManagers/LinuxGSM/pull/4562 to update the Java version.

**Reply by sugarjig** [reply]
Great! Thank you!

**Comment by dino1816** [comment]
> I'm suddenly getting this error when starting a Minecraft server


Hey I don't want to sound like a dick and I LOVEEEE LinuxGSM but I gotta admit that Crafty Controller does MC server hosting perfectly, and it works better.

You have web GUI and you can download, run and manage multiple servers through one interface and it looks pretty polished, I would say it looks enterprise alike.

Just wanted to advise you on trying it out, I have many servers but only MC is not hosted by LGSM since I love Crafty ;)

Many luck to you in the future :P
